reactive/layer_gitlab.py

```diff
@@ -194,3 +198,34 @@ def configure_proxy():
 
     hookenv.status_set("active", HEALTHY)
     set_flag("reverseproxy.configured")
+
+
+def get_runner_token():
+    """Get the runner token for registering a runner."""
+    cmd = (
+        "gitlab-rails runner -e production "
+        "'puts Gitlab::CurrentSettings.current_application_settings.runners_registration_token'"
+    )
+    result = subprocess.check_output(cmd, shell=True)
+    return result.decode("utf-8").rstrip("\n")
+
+
+@when_all("endpoint.runner.joined", "gitlab.configured")
+@when_not("runner.published")
+def publish_runner_config():
+    """Publish the configuration for a runner to register."""
+    endpoint = endpoint_from_flag("endpoint.runner.joined")
+    server_uri = "http://{}".format(socket.getfqdn())
```

Comment:
Should we default to the reverse-proxy configured hostname here if configured, in case runners are not on the same LAN as the server?

Reply:
Great catch, and look at that there's already a helper function to do just that. I'm implementing it now and running through tests.